### PR TITLE
Controller reconciliation rework - part 2 (ServiceBinding)

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/controller/controller_binding_test.go [hunk 1]
```diff
@@ -923,15 +939,10 @@ func TestReconcileServiceBindingDelete(t *testing.T) {
 
 	actions := fakeCatalogClient.Actions()
 	// The actions should be:
-	// 0. Updating the current operation
-	// 1. Updating the ready condition
-	assertNumberOfActions(t, actions, 2)
+	// 0. Updating the ready condition
```
Comment (nilebox) [inline pkg/controller/controller_binding_test.go]: `0.` is probably redundant now that you removed `1.`?

File: pkg/controller/controller_binding_test.go [hunk 2]
```diff
@@ -413,6 +418,23 @@ func TestReconcileServiceBindingWithParameters(t *testing.T) {
...
+			"first-arg",
+			"second-arg",
+		},
+		"name": "test-param",
+	}
+	expectedParametersChecksum := generateChecksumOfParametersOrFail(t, expectedParameters)
+
+	binding = assertServiceBindingOperationInProgressWithParametersIsTheOnlyCatalogAction(t, fakeCatalogClient, binding, v1beta1.ServiceBindingOperationBind, expectedParameters, expectedParametersChecksum)
+	fakeCatalogClient.ClearActions()
+	fakeKubeClient.ClearActions()
```
Comment (nilebox) [inline pkg/controller/controller_binding_test.go]: You clear `fakeKubeClient` actions without checking them (here and in other places). My thinking is:
1. If you don't care about fakeKubeClient actions - don't clear them
2. If you do care then you should assert them before clearing?

Or is it just for making sure that if we ever need to check them in the future, they won't be confusing?

Comment (luksa) [reply pkg/controller/controller_binding_test.go]: Yeah, you're right. The tests should check the actions on all three clients after each call to reconcile(). I'll add the checks.

File: pkg/controller/controller_binding_test.go [hunk 3]
```diff
@@ -76,10 +76,6 @@ func TestReconcileServiceBindingNonExistingServiceInstance(t *testing.T) {
 	assertNumberOfActions(t, actions, 1)
 
 	// There should only be one action that says it failed because no such instance exists.
-	updateAction := actions[0].(clientgotesting.UpdateAction)
-	if e, a := "update", updateAction.GetVerb(); e != a {
-		t.Fatalf("Unexpected verb on actions[0]; %s", expectedGot(e, a))
-	}
```
Comment (luksa) [inline pkg/controller/controller_binding_test.go]: Note that the assertUpdateStatus() in the next line already checks the verb, that's why I deleted these lines

**Pull request merged**